src/Umbraco.Web/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs

```diff
@@ -120,7 +120,9 @@ public override object ConvertIntermediateToObject(IPublishedElement owner, IPub
                             settingsData = null;
                     }
 
-                    var layoutRef = new BlockListItem(contentGuidUdi, contentData, settingGuidUdi, settingsData);
+                    var layoutType = typeof(BlockListItem<,>).MakeGenericType(contentData.GetType(), settingsData?.GetType() ?? typeof(IPublishedElement));
+                    var layoutRef = (BlockListItem)Activator.CreateInstance(layoutType, contentGuidUdi, contentData, settingGuidUdi, settingsData);
```

Comment:
Because `IPublishedModelFactory` only allows creating generic lists, this currently uses basic reflection/activation (`PublishedModelFactory` emits constructor functions and caches them). We might want to do the same here (eventually), because that might gain some performance (especially when lots of items are the same content/settings type).

Reply:
So it's possible to get the actual/CLR model type using `ModelType.For("contentTypeAlias")` and `IPublishedModelFactory`s `MapModelType(modelType)` (see PR https://github.com/umbraco/Umbraco-CMS/pull/8854), but that still doesn't do the caching...